test/snippet/search/bi_fm_index.cpp

```diff
@@ -1,21 +1,19 @@
-#include <vector>
```

Comment:
why was this removed? What about dna4 ?

Reply:
`#include <seqan3/search/fm_index/all.hpp>`
includes this already. 
Even only including `fm_index.hpp` would include the concept, which includes dna4 .. and vector  is included somewhere on the way.

Reply:
Hmmm I completely missed dna4...

I wouldn't think `#include <vector>` is necessary, since all of the nucleotide/aminoacid alphabets come with `alphabet_vector` aliases anyways, and they all already have the vector include.

Reply:
Is the verdict now to include both dna4 and vector, or is it enough to just include dna4?

Reply:
Include both! I would actually like to get rid of the vector includes in the alphabets somehow...

Reply:
Actually you need neither dna nor vector since the bi_fm_index includes them anyway... that's kinda my point.
Why include redundant stuff.

Reply:
Okay, I can include both.

Although I think it isn't very intuitive for people to include vector when they'll be using almost exclusively `seqan3::dna4_vector foo` and not `std::vector<seqan3::dna4> foo`. I would probably assume that `seqan3::dna4_vector` implies that I only need the dna4 include.

Reply:
(Although I now see that in these snippets we are using `std::vector<seqan3::dna4>` and not `seqan3::dna4_vector`.

Reply:
@eseiler Sorry, I previously didn't see your comment or I wouldn't have responded like that.
I still think its correct to fully specify the includes. Unless FM_index is *specified* as providing dna5 you cannot rely on it (changing includes would be a breaking change, see also our stability page).

BTW: why *does* it include dna5?

Reply:
Ok, then lets do `vector` and `dna4` as include.

> BTW: why _does_ it include dna5?

(dna4, right?) It's needed for a concept check, this could probably be replaced by `char`, though...